automation/utils.go

```diff
@@ -0,0 +1,72 @@
+package automation
+
+import (
+	"encoding/json"
+	"errors"
+	"log"
+	"os"
+	"reflect"
+	"time"
+
+	"sigs.k8s.io/yaml"
+)
+
+// Used in TestSuites to create resource objects
+type SetupPaths struct {
+	DbPath        string
+	DbSecretPath  string
+	NdbSecretPath string
+	AppPodPath    string
+	AppSvcPath    string
+}
+
+// CreateTypeFromPath reads a file path, converts it to json, and unmarshals json to a pointer.
+// Ensure that theType is a pointer.
+func CreateTypeFromPath(theType any, path string) (err error) {
+	if theType == nil {
+		return errors.New("theType is nil!")
+	}
+
+	// Check if theType is not a pointer
+	if reflect.ValueOf(theType).Kind() != reflect.Ptr {
+		return errors.New("theType is not a pointer!")
```

Comment:
[Non-Blocking]: Might be a good practice to provide actionable error messages, you can add something like: provide/expected a reference/pointer to a struct type.